src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -39,12 +41,14 @@ public DeleteCommand(int targetUniqueId) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-
-        Person personToDelete = model.getPersonByUniqueId(targetUniqueId);
-        if (targetUniqueId <= 0) {
+        List<Person> lastShownList = model.getFilteredPersonList();
+        Person personToDelete = lastShownList.get(targetUniqueId - 1);
```

Comment:
Yaa, i feel its better do variable checking before we try to access a list with it